.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run pre-commit
```

Comment:
@vickywane how is this integrated with CI to ensure there are not linting conflicts?

Also, what guarantees that `pre-commit` is run on each commit?

Reply:
> Also, what guarantees that pre-commit is run on each commit?

Husky is a battle tested library being used in other large scale application. The generated `.husky/husky.sh` script is always registered within the git scripts and the hook is executed each time a git commit is made,

Reply:
There are currently no linting conflicts from the CI runs within this pull request, and the hook is continuously triggered on each commit against this branch.

Reply:
See comments and changes related to ensuring that git knows about husky.